GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java

```diff
@@ -0,0 +1,135 @@
+package gad.tests;
+
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import gad.binomilia.*;
+
+/**
+ * 
+ * @author Mihir Mahajan
+ * 
+ *         These tests only test if your BinomialHeap/BinomialTree itself works.
+ *         They don't test the logging. Still, they may not work if the logging
+ *         is incorrect!
+ * 
+ *         I can not guarantee the correctness or accuracy of these tests. I
+ *         hope they help, enjoy!
+ */
+
+public class BinomialHeapTester {
+
+	BinomialHeap heap;
+	StudentResult result;
+
+	@BeforeEach
+	void init() {
+
+		heap = new BinomialHeap();
+		result = new StudentResult();
+	}
+
+	@Test
+	void testIllegal() {
+		assertThrows(RuntimeException.class, () -> heap.min(),
+				"Please throw an exception when heap is empty... thats the least u can do");
+		assertThrows(RuntimeException.class, () -> heap.deleteMin(result),
+				"Would u please just throw me an exception >.<");
+	}
+
+	@Test
+	void testMin() {
+		for (int i = -300; i < 300; i++) {
+			heap.insert(i, result);
+		}
+		assertEquals(-300, heap.min(), "Wrong value for min!");
+	}
+
+	@Test
+	void testDeleteMin() {
+		List<Integer> nums = new ArrayList<>();
+		for (int i = -500; i < 500; i++) {
+			heap.insert(i, result);
+			nums.add(i);
+		}
+
+		for (int x : nums) {
+			assertEquals(x, heap.min(), "Wrong value for min!");
+			int deleted = heap.deleteMin(result);
+			assertEquals(x, deleted,
+					"Either did not delete the smallest number or the expected number wasnt in the heap");
+		}
+	}
+
+	@Test
+	void testBigNumbers() {
+		List<Integer> nums = new ArrayList<>();
+		for (int i = -100; i < 1000; i++) {
+			int num = new Random().nextInt();
+			nums.add(num);
+			heap.insert(num, result);
+			int minList = nums.stream().min((x, y) -> Integer.compare(x, y)).get();
+			int minHeap = heap.min();
+			assertEquals(minList, minHeap, "Wrong minimum " + i);
+		}
+		for (int i = 0; i < 500; i++) {
```

Comment:
Because then there would be only one element in the heap, which would get removed per iteration. The errors (at least in my case) occured due to big heaps and bigger numbers.

Reply:
So then why are those loops not connected

Reply:
I see overlooked something.